### PR TITLE
[Doc] Add document about exclusive_cpu_cores

### PR DESCRIPTION
**Pull request by ZiheLiu**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/zh/administration/management/resource_management/resource_group.md [hunk 1]
```diff
@@ -29,25 +30,78 @@ displayed_sidebar: docs
...
+
+    - **限额**：该资源组只能使用这 `exclusive_cpu_cores` 个 CPU cores。即使其他资源组有空闲的 CPU 资源，该资源组也不能使用。
+
+  - 取值范围
+    - 一个资源组的 `exclusive_cpu_cores` 的取值范围为 (0, `min_be_cpu_cores - 1`]，其中 `min_be_cpu_cores` 表示所有 BE 的 CPU 核数的最小值。只有大于 0 时才生效。
+    - `exclusive_cpu_cores` 大于 0 的资源组称为 exclusive 资源组，分配给它的 CPU core 称为 exclusive cores；其余资源组称为 shared 资源组，他们运行在非 exclusive cores 上，称为 shared cores。
+    - 所有资源组的 `exclusive_cpu_cores` 之和不能超过 `min_be_cpu_cores - 1`。之所以最大值为 `min_be_cpu_cores - 1` 而非 `min_be_cpu_cores`，是为了让 shared cores 至少能被分配到一个 CPU core。
+
+  - 与 `cpu_weight` 的关系：
+    - `cpu_weight`  和 `exclusive_cpu_cores` 只能同时并且必须生效一个，因为 exclusive 资源组可以在自己完全拥有的为其预留的 `exclusive_cpu_cores` 个 CPU cores 上运行，无须通过 `cpu_weight` 分配到相对份额的 CPU 时间片。
```
Comment (wangruin) [inline docs/zh/administration/management/resource_management/resource_group.md:76]: `只能同时` 不对吧。直接说：有且仅有一个生效？

File: docs/zh/administration/management/resource_management/resource_group.md [hunk 2]
```diff
@@ -29,25 +30,78 @@ displayed_sidebar: docs
...
+    - 因此，`exclusive_cpu_cores` 和 `cpu_weight` 有且只能有一个为正数。
+
+  - 此外，be.conf 有一个配置项 `enable_resource_group_cpu_borrowing`  来指定是否允许 shared 资源组借用 exclusive 资源组的 exclusive cores。该配置项的值为 true 时，表示允许借用。默认为 true。
+    - 具体来讲，当开启该功能时，
+      - 在一个 BE 上，当一个 exclusive 资源组没有任务运行时，shared 资源组可以暂时借用该 exclusive 资源组的 exclusive cores。
+      - 在一个 BE 上，当该 exclusive 有任务到来后，shared 资源组不可以再借用该 exclusive 资源组的 exclusive cores，需要尽快让出使用的 exclusive cores。这里可能会有一些调度的延迟和开销，所以如果对隔离性要求极强并且允许浪费一定的 CPU，那么可以选择关闭借用功能。
+
+    - 该配置可以运行时动态修改，通过 `update information_schema.be_configs set value = "true" where name= "enable_resource_group_cpu_borrowing"`。
```
Comment (wangruin) [inline docs/zh/administration/management/resource_management/resource_group.md:84]: 默认为true，这里举例子，改成false 是不是好点？
另外我们可以暴露这个用法给客户吗，之前是不是有人误操作，不写where条件，导致集群数据全丢了？

File: docs/zh/administration/management/resource_management/resource_group.md [hunk 3]
```diff
@@ -307,7 +375,9 @@ ALTER RESOURCE GROUP <group_name> DROP ALL;
 
 ### 查看查询命中的资源组
 
-FE 节点 **fe.audit.log** 的 `ResourceGroup` 列和 `EXPLAIN VERBOSE <query>` 的 `RESOURCE GROUP` 列表示特定查询任务最终所匹配的资源组。
+在查询正在运行时，可以通过 `show proc "/current_queries"` 和 `show proc "/global_current_queries"`  的 `ResourceGroup` 列来查看正在运行的查询命中的资源组。
+
+在查询运行结束后，FE 节点 **fe.audit.log** 的 `ResourceGroup` 列和 `EXPLAIN VERBOSE <query>` 的 `RESOURCE GROUP` 列表示特定查询任务最终所匹配的资源组。
 
 - 如果该查询不受资源组管理，那么该列值为空字符串 `""`。
```
Comment (wangruin) [inline docs/zh/administration/management/resource_management/resource_group.md:382]: 这里2行合成一句会不会好点，现在分了2行，结果都说的空字符串。改成
如果`ResourceGroup`列和`RESOURCE GROUP` 列的值为空字符串`""`，表示该查询不受资源组管理或者受资源组管理，但是没有匹配到分类器。

File: docs/zh/administration/management/resource_management/resource_group.md [hunk 4]
```diff
@@ -381,6 +451,51 @@ MySQL [(none)]> SHOW USAGE RESOURCE GROUPS;
...
+| 10223 | pip_exec_com | 2095248        | 10         |
+| 10223 | pip_exec_com | 2098745        | 10         |
+| 10223 | pip_exec_com | 2085338        | 10         |
+| 10223 | pip_exec_com | 2101221        | 10         |
+| 10223 | pip_exec_com | 2093901        | 10         |
+| 10223 | pip_exec_com | 2092364        | 10         |
+| 10223 | pip_exec_com | 2091366        | 10         |
++-------+--------------+----------------+------------+
+```
+
```
Comment (wangruin) [inline docs/zh/administration/management/resource_management/resource_group.md:496]: 多余的空格

File: docs/zh/administration/management/resource_management/resource_group.md [hunk 5]
```diff
@@ -6,20 +6,21 @@ displayed_sidebar: docs
 
 本文介绍如何使用资源隔离功能。
 
-自 2.2 版本起，StarRocks 支持资源组管理，集群可以通过设置资源组（Resource Group）的方式限制查询对资源的消耗，实现多租户之间的资源隔离与合理利用。在 2.3 版本中，StarRocks 支持限制大查询，集群可以进一步控制大查询对资源的消耗，避免少数的大查询耗尽系统资源，进而影响系统稳定性。StarRocks 2.5 版本支持通过资源组对导入计算进行资源隔离，从而间接控制导入任务对集群资源的消耗。
+自 2.2 版本起，StarRocks 支持资源组管理，集群可以通过设置资源组（Resource Group）的方式限制查询对资源的消耗，实现多租户之间的资源隔离与合理利用。在 2.3 版本中，StarRocks 支持限制大查询，集群可以进一步控制大查询对资源的消耗，避免少数的大查询耗尽系统资源，进而影响系统稳定性。StarRocks 2.5 版本支持通过资源组对导入计算进行资源隔离，从而间接控制导入任务对集群资源的消耗。StarRocks 3.3.4 版本支持对 CPU 资源进行硬隔离。
```
Comment (wangruin) [inline docs/zh/administration/management/resource_management/resource_group.md:9]: 3.3.5

File: docs/zh/administration/management/resource_management/resource_group.md [hunk 6]
```diff
@@ -29,25 +30,78 @@ displayed_sidebar: docs
...
-  
-  > 说明：例如，在 16 核的 BE 节点中设置三个资源组 rg1、rg2、rg3，`cpu_core_limit` 分别设置为 `2`、`6`、`8`。当在该 BE 节点满载时，资源组 rg1、rg2、rg3 能分配到的 CPU 核数分别为 BE 节点总 CPU 核数 ×（2/16）= 2、 BE 节点总 CPU 核数 ×（6/16）= 6、BE 节点总 CPU 核数 ×（8/16）= 8。如果当前 BE 节点资源非满载，rg1、rg2 有负载，rg3 无负载，则 rg1、rg2 分配到的 CPU 核数分别为 BE 节点总 CPU 核数 ×（2/8）= 4、 BE 节点总 CPU 核数 ×（6/8）= 12。
-  
-- `mem_limit`：该资源组在当前 BE 节点可使用于查询的内存（query_pool）占总内存的百分比（%）。取值范围为 (0,1)。
+
+您可以为资源组设置如下的资源限制。
+
+| 配置名称                   | 描述                                                      | 取值范围                                                     | 默认值 |
+| -------------------------- | --------------------------------------------------------- | ------------------------------------------------------------ | ------ |
+| cpu_weight                 | 该资源组在一个 BE 节点上调度的权重。                      | (0, `avg_be_cpu_cores`] (大于 0 时生效，`cpu_weight` 和 `exclusive_cpu_cores` 有且只能有一个为正数) | 0      |
```
Comment (wangruin) [inline docs/zh/administration/management/resource_management/resource_group.md:39]: 个人感觉是不是可以不加 ``cpu_weight` 和 `exclusive_cpu_cores` 有且只能有一个为正数) `，下面有几个地方都强调了，这里保持表格简洁

File: docs/zh/administration/management/resource_management/resource_group.md (hunk 3 above)
Comment (ZiheLiu) [reply docs/zh/administration/management/resource_management/resource_group.md:382]: 两行可能会清晰一些，来区分正在运行时和运行结束后，查看的方式。
在下面加了一行`对于上述命中显示资源组信息：`。

File: docs/zh/administration/management/resource_management/resource_group.md (hunk 2 above)
Comment (ZiheLiu) [reply docs/zh/administration/management/resource_management/resource_group.md:84]: 有道理，举例子改成 `false` 了。

Comment (ZiheLiu) [reply docs/zh/administration/management/resource_management/resource_group.md:84]: 我看文档中其他地方也有使用 `update information_schema.be_configs set value` 的例子。https://docs.starrocks.io/zh/docs/3.2/using_starrocks/block_cache/#%E6%9F%A5%E7%9C%8B-data-cache-%E7%8A%B6%E6%80%81

**Review by EsoragotoSpirit — DISMISSED** [review]

**Review by EsoragotoSpirit — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**